orchestra/models/core/models.py

```diff
@@ -293,6 +307,65 @@ class Role:
     staffing_priority = models.IntegerField(default=0)
 
 
+class WorkerAvailability(WorkerAvailabilityMixin, BaseModel):
+    """
+    WorkerAvailability describes a worker's availabilty for work
+    in a given week.
+
+    Attributes:
+        worker (orchestra.models.Worker):
+            The worker who is available during this week.
+        week (datetime.date):
+            The week during which the worker is available. Availability is
+            active for a 7-day period following the date stored in this field.
+        hours_available_mon (float):
+            The number of hours the worker is available to work on Monday of
+            the week.
+        hours_available_tues (float):
+            The number of hours the worker is available to work on Tuesday of
+            the week.
+        hours_available_wed (float):
+            The number of hours the worker is available to work on Wednesday of
+            the week.
+        hours_available_thurs (float):
+            The number of hours the worker is available to work on Thursday of
+            the week.
+        hours_available_fri (float):
+            The number of hours the worker is available to work on Friday of
+            the week.
+        hours_available_sat (float):
+            The number of hours the worker is available to work on Saturday of
+            the week.
+        hours_available_sun (float):
+            The number of hours the worker is available to work on Sunday of
+            the week.
+    """
+
+    worker = models.ForeignKey(Worker, on_delete=models.CASCADE,
+                               related_name='availabilities')
+
+    # The Monday (day 0) of the week these preferences apply for.
+    # Default computes the Monday of the current week.
+    week = models.DateField(default=first_day_of_the_week)
+
+    # Hours available for each day of the week
+    hours_available_mon = models.FloatField(default=0, verbose_name='Monday')
+    hours_available_tues = models.FloatField(default=0, verbose_name='Tuesday')
+    hours_available_wed = models.FloatField(
+        default=0, verbose_name='Wednesday')
+    hours_available_thurs = models.FloatField(
+        default=0, verbose_name='Thursday')
+    hours_available_fri = models.FloatField(default=0, verbose_name='Friday')
+    hours_available_sat = models.FloatField(default=0, verbose_name='Saturday')
+    hours_available_sun = models.FloatField(default=0, verbose_name='Sunday')
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
```

Comment:
is this how uniqueness works in newer versions of Django? i vaguely remember unique_together, but this looks more elegant tbh :)

Reply:
it's the future!